beta/nncf/tensorflow/graph/converter.py

```diff
@@ -190,3 +143,143 @@ def get_graph_to_layer_var_names_map(concrete_fun):
                 nxmodel.add_edge(input_node_name, node_name)
 
     return nxmodel
+
+
+def convert_keras_model_to_nncf_graph(model: tf.keras.Model) -> NNCFGraph:
+    """
+    Convert Keras model graph to the NNCFGraph
+
+    :param model: Keras model
+    :return: NNCFGraph
+    """
+    func_model = is_functional_model(model)
+    seq_model = is_sequential_model(model)
+
+    if not func_model and not seq_model:
+        RuntimeError('convert_keras_model_to_nxmodel function supports '
+                     'only sequential or functional models')
+
+    if func_model:
+        nncf_graph = _get_nncf_graph_from_functional(model)
+    else:
+        nncf_graph = _get_nncf_graph_from_sequential(model)
+
+    return nncf_graph
+
+
+def _get_nncf_graph_from_functional(model: tf.keras.Model) -> NNCFGraph:
+    model_config = model.get_config()
+    raw_nodes = _prepare_raw_nodes(model)
+    return _get_nncf_graph_from_raw_nodes(model_config, raw_nodes)
+
+
+def _prepare_raw_nodes(model: tf.keras.layers.Layer) -> Dict:
+    model_config = model.get_config()
+    raw_nodes = Dict()
+    for layer in model_config['layers']:
+        layer_name = layer['name']
+        layer_type = _get_layer_type(layer)
+        layer_dtype = _get_layer_dtype(layer)
+        data_format = layer['config'].get('data_format')
+        if layer['inbound_nodes']:
+            is_shared = len(layer['inbound_nodes']) > 1
+            for i, inbound_node in enumerate(layer['inbound_nodes']):
+                instance = raw_nodes[layer_name][i]
```

Comment:
@alexsu52 I have a code style question. Since addict.Dict supports both dict['a'] and dict.a addressing, which one is preferable? I've seen both usages throughout the tool.